### PR TITLE
Add alice-ai extension

### PR DESCRIPTION
**Pull request by wojciechkrol**
## Description

Alice AI is a daily productivity tool that uses your selection to execute actions using OpenAI's GPT models. It allows you to easily browse your actions, execute them, and go back to your result history. You can also add your own actions, create quicklinks to them, and back them up.

## Screencast

https://github.com/raycast/extensions/assets/14108143/b28702da-b98f-4e4c-a6c9-36a6f634e5b9

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by wojciechkrol** [comment]
In the last few days, I made a few corrections to the extension. I think it's ready for review now :)

**Comment by pernielsentikaer** [comment]
Thanks for your contribution 🔥

What's the difference between the and extensions like https://www.raycast.com/hakob_nersesian/grammari-x or https://www.raycast.com/alanzchen/chatgpt-quick-actions 🙂

**Comment by wojciechkrol** [comment]
Hi @pernielsentikaer,

Thank you for reviewing my Pull Request. The significant difference between the extensions you mentioned is the configuration and flexibility in operation. Alice AI allows for creating custom actions useful in daily work. GrammariX focuses on text improvement, while ChatGPT Quick Actions, I must admit I tried using it but was annoyed by the inability to create custom actions (only predefined ones are available). Additionally, in my solution, I provide the ability to create QuickLink actions so they can be called directly from the main Raycast window. Alice primarily focuses on input from selection - the idea is to quickly process the selected text with a chosen action.

Rewriting text, summarizing a selected article, or translating from one language to another are just a few possibilities. I often create actions specialized in a particular topic, and the fact that I can quickly call them from within Raycast is a huge advantage.

**Comment by pernielsentikaer** [comment]
What about enhancing the current one (ChatGPT Quick Actions) with the missing features so we don't end up with a lot of extensions with the same purpose?

**Comment by wojciechkrol** [comment]
The current extension operates on completely different assumptions, and improving it would necessitate a brand-new codebase, essentially mirroring what I am proposing. Furthermore, I am hesitant to disrupt the creator's vision, as they may not be in agreement with my suggestions. Another concern is the level of control over the extension. By working on this project independently, I have complete authority over its development, enabling me to introduce new features and address bugs without being dependent on the author's approval.

**Comment by pernielsentikaer** [comment]
Can you look into this error which crashes Raycast, do you think you can handle it gracefully?

alice-ai 2024-04-17 at 09 14 25

I want to give ti a spin so I can compare it 🙂

**Comment by wojciechkrol** [comment]
Hello @pernielsentikaer,

Whoops, I forgot that error handling in communication would be useful :) Thank you.

The fix has already been implemented.

**Comment by pernielsentikaer** [comment]
Instead of always showing the hand as icon, maybe do something dynamic like I did here
Raycast17042024-1ZTJfCMl png@2x

You can check https://www.raycast.com/pernielsentikaer/remember-the-date 🙂

**Comment by wojciechkrol** [comment]
That's a good idea 👍🏻

**Comment by wojciechkrol** [comment]
@pernielsentikaer added colors and different icons to the list, and I made some improvements to the initial version of the extension. I also included a "favorites" feature, which I had added for myself a few days ago.

I didn't add any migrations, so please clear the Local Storage extension.

**Comment by pernielsentikaer** [comment]
That's awesome. Would it make sense to add the possibility to sort favorites while we're in the initial state too?

File: extensions/alice-ai/src/commands.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export default function ActionsCommand(props: LaunchProps<{ arguments: Arguments }>) {
+  const ready = useActionsAreReady();
+
+  if (!ready) {
+    return <Detail />;
+  }
+
+  if (props.arguments.id !== undefined) {
+    return <CommandExecute id={props.arguments.id} />;
+  }
```
Comment (pernielsentikaer) [inline extensions/alice-ai/src/commands.tsx:20]: We can maybe remove all this and just send ready={ready} to CommandList which as `isLoading` as property instead

Comment (wojciechkrol) [reply extensions/alice-ai/src/commands.tsx:20]: The extension's state is persisted to LocalStorage, and every time the command is called, data is rehydrated from LocalStorage to the state. Unfortunately, this process is asynchronous, and I need to wait for the rehydrate process to finish before I can use the data (my useActionsAreReady hook).

In this command, I have two responsibilities:
1. Displaying the list of actions - which is default.
2. Immediately proceeding to execute an action by calling the command created using "Create QuickLink".

In both cases, the store must already be loaded to handle the request correctly. If I moved the loading to CommandList, I would skip CommandExecute, which might not work due to the extension state not being loaded.

**Comment by wojciechkrol** [comment]
> That's awesome. Would it make sense to add the possibility to sort favorites while we're in the initial state too?

What do you mean? All actions are sorted by action name by default. Actions added to favorites are separated in the "Favorites" section at the top of the list and sorted in the same way. Other actions are in the "Actions" section, below "Favorites".

Initially, there are no favorite actions defined.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by wojciechkrol** [comment]
@pernielsentikaer, thank you for the suggestions and cooperation!

**Pull request merged**